### PR TITLE
Removing both finalizers in federation controllers in a single update

### PR DESCRIPTION
**Pull request by nikhiljindal**
Fixes https://github.com/kubernetes/kubernetes/issues/43828

There is a bug right now where the controller fails to delete the object if one finalizer is removed and the second isnt.
This updates the code so that both the finalizers are removed in a single API call. Kept the code changes minimum to enable cherrypick in 1.6.x

cc @csbell @kubernetes/sig-federation-bugs 

```release-note
Fix for [failure to delete federation controllers with finalizers](https://github.com/kubernetes/kubernetes/issues/43828).
```

### CODE REVIEW

**Comment by marun** [comment]
I think it would be preferable to use sets.String rather than a slice, but I guess that would be even more work.

File: federation/pkg/federation-controller/configmap/configmap_controller.go [hunk 1]
```diff
@@ -202,13 +203,13 @@ func (configmapcontroller *ConfigMapController) hasFinalizerFunc(obj pkgruntime.
...
-// removeFinalizerFunc removes the finalizer from the given objects ObjectMeta. Assumes that the given object is a configmap.
-func (configmapcontroller *ConfigMapController) removeFinalizerFunc(obj pkgruntime.Object, finalizer string) (pkgruntime.Object, error) {
+// removeFinalizerFunc removes the given finalizers from the given objects ObjectMeta. Assumes that the given object is a configmap.
+func (configmapcontroller *ConfigMapController) removeFinalizerFunc(obj pkgruntime.Object, finalizers []string) (pkgruntime.Object, error) {
 	configmap := obj.(*apiv1.ConfigMap)
 	newFinalizers := []string{}
 	hasFinalizer := false
 	for i := range configmap.ObjectMeta.Finalizers {
-		if string(configmap.ObjectMeta.Finalizers[i]) != finalizer {
+		if !sliceutils.ContainsString(finalizers, configmap.ObjectMeta.Finalizers[i]) {
```
Comment (csbell) [inline federation/pkg/federation-controller/configmap/configmap_controller.go]: I'd move this as a utility in deletion_helper.go.
deletion_helper.HasFinalizer(finalizers, configmap.ObjectMeta.Finalizers[i])
Clarifies the code and removes the util/slice dependency in all the callers.

Comment (nikhiljindal) [inline federation/pkg/federation-controller/configmap/configmap_controller.go]: Moved ContainsString to deletionhelper/util.go.

I am working on another PR to remove all this duplication.

**Comment by nikhiljindal** [comment]
@k8s-bot non-cri e2e test this

**Comment by nikhiljindal** [comment]
Verified that federation e2e tests pass.
This is now good to go.
Adding LGTM as per approval from @csbell 

/lgtm

**Pull request merged**